gtsfm/bundle/bundle_adjustment.py

```diff
@@ -177,13 +198,16 @@ def __optimize_factor_graph(self, graph: NonlinearFactorGraph, initial_values: V
     def run(
         self,
         initial_data: GtsfmData,
+        absolute_pose_priors: List[Optional[PosePrior]],
+        relative_pose_priors: Dict[Tuple[int, int], Optional[PosePrior]],
```

Comment:
I think since we decided on adding the IMU pose variables in bundle adjustment, we will need another input for the hard constraint between imu and camera. I think we can call it fixed_body_T_camera? 
Can we add those in the same PR? Might have to modify loader as well.

Reply:
Do you propose adding it as a third argument?

Reply:
Yes

Reply:
it would be nice if we could indicate that these priors are in the body frame, not the camera frame. Or maybe leave this change to the bundle adjustment PR and only plumb the prior until MVO? 

I was not thinking of the two-view BA use case, only the final BA use. So it would be tricky. In two-view BA, the relative pose prior will be between cameras and in final BA between IMU's? Or will we change two-view BA to include IMU poses as well?

Reply:
Yeah its a bit tricky. Let's list down what priors are to be used:

## 2-view BA:
- prior on i2Ti1: available as relative_pose_priors. Hence, relative pose priors are always between cameras.


## BA:
- prior/hard-constraint between camera and IMU
- relative constraints between IMU.

I think we need a way of representing IMU in the BA graph, maybe write something specific to hilti.

Reply:
Yeah, I think have another class for rig BA might be better, let us discuss that in the BA PR. Thanks for making the note here!